### PR TITLE
Fix correct drawing when elements are skipped and overlaid

### LINKED ISSUE #319: Draw order incorrect if empty (null) values in dataset
When applying data labels to a radar chart, the z-index and draw order becomes out of sync if there are null/empty values in the data set.

image

### PR DESCRIPTION
**Pull request by stockiNail**
Fix #319

### CODE REVIEW

File: test/fixtures/drawing/draw-correct-order-if-undefined-values.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// https://github.com/chartjs/chartjs-plugin-datalabels/issues/319
```
Comment (simonbrunel) [inline test/fixtures/drawing/draw-correct-order-if-undefined-values.js:1]: You are testing `null` values, not `undefined` so I would suggest to rename these test files as: 

- `draw-skipped-values-in-correct-order.js`
- `draw-skipped-values-in-correct-order.png`

File: test/fixtures/drawing/draw-correct-order-if-undefined-values.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// https://github.com/chartjs/chartjs-plugin-datalabels/issues/319
+
+export default {
```
Comment (simonbrunel) [inline test/fixtures/drawing/draw-correct-order-if-undefined-values.js:3]: If you look at the other tests, there are very minimal and focus on what need to be tested. That's to improve test stability across different platforms but also when upgrading Chart.js. So I would suggest to get rid the options that doesn't bring any value:

- make label square (i.e. no border radius)
- hide the label text (i.e. `{ font: { size: 0 } }`)
- hide lines between data points

At the end, the generated PNG should only show red squares on top of blue squares

File: test/fixtures/drawing/draw-correct-order-if-undefined-values.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          padding: 10,
+          borderRadius: 20
+        }
+      }
+
+    }
+  },
+  options: {
+    canvas: {
+      height: 512,
```
Comment (simonbrunel) [inline test/fixtures/drawing/draw-correct-order-if-undefined-values.js:56]: 256px x 256px PNG should be enough?

**Review by simonbrunel — CHANGES_REQUESTED** [review]

File: test/fixtures/drawing/draw-correct-order-if-undefined-values.js (hunk 2 above)
Comment (stockiNail) [reply test/fixtures/drawing/draw-correct-order-if-undefined-values.js]: done

File: test/fixtures/drawing/draw-correct-order-if-undefined-values.js (hunk 3 above)
Comment (stockiNail) [reply test/fixtures/drawing/draw-correct-order-if-undefined-values.js]: done

File: test/fixtures/drawing/draw-correct-order-if-undefined-values.js (hunk 1 above)
Comment (stockiNail) [reply test/fixtures/drawing/draw-correct-order-if-undefined-values.js]: done

**Comment by stockiNail** [comment]
@simonbrunel I have applied the review you suggested

**Review by simonbrunel — APPROVED** [review]

**Comment by simonbrunel** [comment]
Thanks @stockiNail! This PR has been manually merged (fed9762087d9224d12f87f4360dc7dec3abab5a1) since I wanted the test to use bigger labels. I'm not sure why it wasn't enabled for this PR but make sure to check the option to allow maintainers to push to your repository / branch.

**Comment by stockiNail** [comment]
> Thanks @stockiNail! This PR has been manually merged ([fed9762](https://github.com/chartjs/chartjs-plugin-datalabels/commit/fed9762087d9224d12f87f4360dc7dec3abab5a1)) since I wanted the test to use bigger labels. I'm not sure why it wasn't enabled for this PR but make sure to check the option to allow maintainers to push to your repository / branch.

It is weird. Usually I fetch al update a from the main before starting a PR.
Anyway thank uou, I’ll pay more attention next time

**Comment by simonbrunel** [comment]
I'm thinking about the options ["Allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests) when [creating the PR (see step 7)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).

Screenshot 2022-07-16 at 19 40 20

**Pull request closed without merging**